From 60f7d99defa6dd9a9b614c82bedabef34b1d9c1d Mon Sep 17 00:00:00 2001
From: libin <bin.li@rock-chips.com>
Date: Sun, 28 Apr 2019 19:57:08 +0800
Subject: [PATCH 1/2] AFBC : HWC need to enable AFBC encoding, if the primary
 CRTC supports AFBC encoding.if not to disable AFBC encoding.

Change-Id: Idd19709ec2973f2b5bcb0b8d0d7aa47443250402
Signed-off-by: libin <bin.li@rock-chips.com>
---
 drmresources.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drmresources.cpp b/drmresources.cpp
index 30bda7d..80a0eef 100755
--- a/drmresources.cpp
+++ b/drmresources.cpp
@@ -900,6 +900,20 @@ int DrmResources::UpdateDisplayRoute(void)
     }
   }
 
+  if(primary && primary->encoder() && primary->encoder()->crtc())
+  {
+    DrmCrtc *crtc = primary->encoder()->crtc();
+    if(crtc->get_afbc()){
+      property_set( PROPERTY_TYPE ".gralloc.disable_afbc", "0");
+      ALOGD_IF(log_level(DBG_VERBOSE), "%s:line=%d primary conn[%d] crtc=%d support AFBC(%d), to use AFBC\n",
+               __FUNCTION__, __LINE__, primary->id(), crtc->id(),crtc->get_afbc());
+    }else{
+      property_set( PROPERTY_TYPE ".gralloc.disable_afbc", "1");
+      ALOGD_IF(log_level(DBG_VERBOSE), "%s:line=%d primary conn[%d] crtc=%d support AFBC(%d), to disable AFBC\n",
+               __FUNCTION__, __LINE__, primary->id(), crtc->id(),crtc->get_afbc());
+    }
+  }
+
   if(primary && !strcmp(connector_type_str(primary->get_type()), "HDMI-A"))
   {
     if (primary && primary->encoder() && primary->encoder()->crtc())
-- 
2.7.4

